agent/functional_tests/tests/functionaltests_windows_test.go

```diff
@@ -344,3 +356,345 @@ func TestTwoTasksSharedLocalVolume(t *testing.T) {
+                                                                      }
+                                                                  ]
+                              }
+}`
+
+	_, err := svc.Upload(&s3manager.UploadInput{
+		Bucket: aws.String(s3Bucket),
+		Key:    aws.String("gmsa-test-file.json"),
+		Body:   strings.NewReader(testCredSpecData),
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == errCodeAccessDenied {
+			t.Skip("Skipping the test since lacking necessary s3 access permission")
+		} else {
+			t.Fatalf("Unable to upload config file to s3 which is needed for the test: %v", err)
+		}
+	}
+
+	// Setup agent
+	RequireDockerAPIVersion(t, ">=1.40")
+
+	agentOptions := AgentOptions{
+		ExtraEnvironment: map[string]string{
+			"ZZZ_SKIP_DOMAIN_JOIN_CHECK_NOT_SUPPORTED_IN_PRODUCTION": "true",
+			"ECS_ENGINE_TASK_CLEANUP_WAIT_DURATION":                  "1m",
+		},
+	}
+
+	agent := RunAgent(t, &agentOptions)
+	defer agent.Cleanup()
+
+	agent.RequireVersion(">=1.33.0")
+
+	// Setup taskdef overrides
+	tdOverrides := make(map[string]string)
+	tdOverrides["$$$TEST_REGION$$$"] = aws.StringValue(ECS.Config.Region)
+	tdOverrides["$$$EXECUTION_ROLE$$$"] = executionRole
+	tdOverrides["$$$TEST_S3_BUCKET$$$"] = s3Bucket
+
+	testTask, err := agent.StartTaskWithTaskDefinitionOverrides(t, "gmsa-s3", tdOverrides)
+	require.NoError(t, err)
+
+	// Wait for the container to start
+	testTask.WaitRunning(waitTaskStateChangeDuration)
+	taskID, err := GetTaskID(aws.StringValue(testTask.TaskArn))
+	require.NoError(t, err)
+	assert.NotEmpty(t, taskID)
+
+	// Verify container metadata
+	cid, err := agent.ResolveTaskDockerID(testTask, "windows_sample_app")
+	require.NoError(t, err, "Error resolving docker id for container in task")
+	assert.NotEmpty(t, cid)
+
+	// Setup docker client to validate credentialspec
+	endpoint := utils.DefaultIfBlank(os.Getenv(dockerEndpointEnvVariable), dockerEndpoint)
+	client, _ := sdkClient.NewClientWithOpts(sdkClient.WithHost(endpoint), sdkClient.WithVersion(sdkclientfactory.GetDefaultVersion().String()))
+
+	// credentialspec=file://s3_c3ce6880-f815-4044-af03-29f934836293_gmsa-test-file.json
+	testCredentialSpecOption := fmt.Sprintf("credentialspec=file://s3_%s_gmsa-test-file.json", taskID)
+
+	err = verifyContainerCredentialSpec(client, cid, testCredentialSpecOption)
+	assert.NoError(t, err)
+
+	// Stop test task
+	testTask.Stop()
+
+	err = testTask.WaitStopped(waitTaskStateChangeDuration)
+	assert.NoError(t, err)
+
+	// Cleanup the test file
+	testCredSpecFilePath := fmt.Sprintf("C:/ProgramData/docker/credentialspecs/s3_%s_gmsa-test-file.json", taskID)
+	err = os.Remove(testCredSpecFilePath)
+	assert.NoError(t, err)
+}
+
+func verifyContainerCredentialSpec(client *sdkClient.Client, id, credentialspecOpt string) error {
+	dockerContainer, err := client.ContainerInspect(context.TODO(), id)
+	if err != nil {
+		return err
+	}
+
+	for _, opt := range dockerContainer.HostConfig.SecurityOpt {
```

Comment:
nit: Though it's a list, docker doesn't support multiple credentialspec.

Reply:
N/A as we only permit one credentialspec per container.